apps/react-playground/src/app/Widgets/MessagingPrototype/services/messaging-app.service.tsx

```diff
@@ -0,0 +1,163 @@
+  deso: Deso,
+  setDerivedResponse: (
+    derivedResponse: Partial<DerivedPrivateUserInfo>
+  ) => void,
+  setHasSetupAccount: (hasSetup: boolean) => void
+): Promise<false | Partial<DerivedPrivateUserInfo>> => {
+  let key = deso.identity.getUserKey();
+  if (!key) {
+    // are they already logged in? if not prompt them
+    const res = await login(deso);
+    key = res.key;
+    if (!key) {
+      alert('failed to login');
+      return false;
+    }
+  }
+  const userResponse = await deso.user.getUsersStateless({
+    PublicKeysBase58Check: [key],
+    SkipForLeaderboard: true,
+    IncludeBalance: true,
+  });
+  const user = userResponse?.UserList?.[0];
+  if (user && user.BalanceNanos === 0) {
+    // does the user have a balance? If not let them know since they will not be able to proceed
+    const openFreeDeso = window.confirm(
+      `no deso funds found for ${key}. click okay to add some through phone verification. Otherwise you can send deso from another account`
+    );
+    if (openFreeDeso) {
+      await deso.identity.phoneVerification('4', undefined, {
+        publicKey: deso.identity.getUserKey() as string,
+      });
+    } else {
+      return false;
+    }
+  } else if (!user) {
+    alert('unable to find user');
+    return false;
+  }
+  let derivedResponse: Partial<DerivedPrivateUserInfo> =
+    getDerivedKeyResponse(key); // does the derived key exist in storage already?
+  if (!derivedResponse.derivedPublicKeyBase58Check) {
+    // if not request one
+    derivedResponse = await requestDerivedKey(deso);
+  }
+  if (!derivedResponse.derivedPublicKeyBase58Check) {
+    alert('failed to authorize derive key');
+    return false;
+  }
+  setDerivedKeyResponse(derivedResponse, key);
+  await authorizeDerivedKey(deso, derivedResponse);
+  await delay(3000); // lazy way to make sure the transaction has had enough time to broadcast,
+  // alternatively you call can the get-txn endpoint to verify it has been broadcasted;
+  const messagingKey = await generateDefaultKey(deso, derivedResponse);
+  if (messagingKey) {
+    setDefaultKey(messagingKey);
+  }
+  setDerivedResponse(derivedResponse);
+  setHasSetupAccount(true);
+
+  return derivedResponse;
+};
+
+export const getConversations = async (
+  deso: Deso,
+  derivedResponse: Partial<DerivedPrivateUserInfo>,
+  setGetUsernameByPublicKeyBase58Check: (publicKey: {
+    [key: string]: string;
+  }) => void,
+  setConversations: (conversations: DecryptedResponse) => void,
+  setSelectedConversationPublicKey: (publicKey: string) => void
+) => {
+  try {
+    if (!derivedResponse) {
+      alert('derived call failed');
+      return {};
+    }
+    let res = await getConversationsMapAndPublicKeyToUsernameMapping(
+      deso,
+      derivedResponse
+    );
+    let conversationsArray = Object.keys(res.conversationMap);
+    setGetUsernameByPublicKeyBase58Check(res.publicKeyToUsername);
+    if (conversationsArray.length === 0) {
+      await deso.utils.encryptMessageV3(
```

Comment:
this is actually a function that encrypts and sends message, so we should probably name it as such.